feat: add `log-metrics` command feature

fukusuket opened the pull request:
## What Changed

- Closed #1474 

## Evidence
### Integration-Test
- https://github.com/Yamato-Security/hayabusa/actions/runs/11794771512

fukusuket commented:
I created prototype! The following features have not yet been implemented🤔
- Timeformat
- Filtering
- Abbreviation 

@YamatoSecurity 
I would appreciate your comments when you have time🙏(Sort order of output results, column order, etc.... etc.)
#### Terminal Output
<img width="1440" alt="スクリーンショット 2024-11-12 12 31 51" src="https://github.com/user-attachments/assets/27de7832-e7e2-4d02-993d-27a9f33be901">

#### CSV Output
<img width="1440" alt="スクリーンショット 2024-11-12 12 33 31" src="https://github.com/user-attachments/assets/37e30c25-3b35-419a-ab26-23b1743d2952">
<img width="1440" alt="スクリーンショット 2024-11-12 12 33 26" src="https://github.com/user-attachments/assets/ac1f85d8-f991-4bf9-8447-1d5468f806c0">

fukusuket commented:
The following two issues may be better separated as separate issues and will be addressed in a separate pull request.
- https://github.com/Yamato-Security/hayabusa/issues/1485
- https://github.com/Yamato-Security/hayabusa/issues/1486

YamatoSecurity commented:
@fukusuket Thanks! Looking good. Just two things about the table besides what you mentioned.
1. Can you rename "Event Count" to just "Events" because this column title is long it is making the column width wider than necessary.
2. Can you separate the Computers, Channels and Providers by newline instead of commas when there are multiple results? (Make them multi-lined) I think this will be easier to read than separating by commas as well as might make the width narrower.

YamatoSecurity commented:
When writing to CSV file, maybe we should separate by the broken pipe character like we do for Hayabusa results?

YamatoSecurity commented:
One more bug I noticed is that the progress bar does not reach 100% for some reason.
<img width="1377" alt="Screenshot 2024-11-12 at 15 51 42" src="https://github.com/user-attachments/assets/8d56ad6b-05f9-4d6b-a842-2f5f75137a48">

fukusuket commented:
@YamatoSecurity 
Thank you for comment! I updated following points! Could you check it?🙏
- https://github.com/Yamato-Security/hayabusa/pull/1484#issuecomment-2469730569
- https://github.com/Yamato-Security/hayabusa/pull/1484#issuecomment-2469731520

Also, I implemented Timeformat/Filter.

fukusuket commented:
```
fukusuke@fukusukenoMacBook-Air hayabusa-2.19.0-mac-aarch64 % ./hayabusa log-metrics -d ../all-evtx/Logs_Win11 --include-computer WIN-KPO4DDU11AB --ISO-8601 -q
Start time: 2024/11/12 18:57

Total event log files: 355
Total file size: 1.8 GB

Currently scanning for log metrics. Please wait.

[00:00:08] 316 / 355 ⠁ [===================================>    ] 89%

"../all-evtx/Logs_Win11/Microsoft-Windows-WinINet%4Operational.evtx"                                                                           ╭──────────────────────┬─────────────────┬────────┬──────────────────────┬──────────────────────┬──────────────────────┬──────────────────────╮
│ Filename             ┆ Computers       ┆ Events ┆ First Timestamp      ┆ Last Timestamp       ┆ Channels             ┆ Providers            │
╞══════════════════════╪═════════════════╪════════╪══════════════════════╪══════════════════════╪══════════════════════╪══════════════════════╡
│ Microsoft-Windows-Ap ┆ WIN-KPO4DDU11AB ┆ 610    ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:43:02. ┆ MS-Win-AppXDeploySvr ┆ MS-Win-AppXDeploy-Sv │
│ pXDeploymentServer%4 ┆                 ┆        ┆ 260472Z              ┆ 939883Z              ┆ /Op                  ┆ r                    │
│ Operational.evtx     ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ke ┆ WIN-KPO4DDU11AB ┆ 363    ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:09. ┆ MS-Win-Kern-PnP/Conf ┆ MS-Win-Kern-PnP      │
│ rnel-PnP%4Configurat ┆                 ┆        ┆ 622783Z              ┆ 408812Z              ┆ ig                   ┆                      │
│ ion.evtx             ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Application.evtx     ┆ WIN-KPO4DDU11AB ┆ 259    ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ App                  ┆ MS-Win-Sec-SPP       │
│                      ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆                      ┆ MS-Win-User Profs    │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ Svc                  │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-WMI           │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ edgeupdate           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-St ┆ WIN-KPO4DDU11AB ┆ 193    ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:09. ┆ MS-Win-StateReposito ┆ MS-Win-StateReposito │
│ ateRepository%4Opera ┆                 ┆        ┆ 622783Z              ┆ 408812Z              ┆ ry/Op                ┆ ry                   │
│ tional.evtx          ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Security.evtx        ┆ WIN-KPO4DDU11AB ┆ 162    ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ Sec                  ┆ Sec                  │
│                      ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ System.evtx          ┆ WIN-KPO4DDU11AB ┆ 137    ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:21. ┆ System               ┆ BTHUSB               │
│                      ┆                 ┆        ┆ 620626Z              ┆ 469910Z              ┆                      ┆ EventLog             │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-DHCPv6-Cli    │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-Dhcp-Cli      │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-Dir-Svcs-SAM  │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-FilterMgr     │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-HAL           │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-Kern-Boot     │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-Kern-General  │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-Kern-Processo │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ r-Pwr                │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-Kern-Pwr      │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-Ntfs          │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-Setup         │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-TPM-WMI       │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-UserModePwrSv │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ c                    │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ MS-Win-Wininit       │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ SCM                  │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ TPM                  │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ User32               │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ Virtual Disk Svc     │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ Workstation          │
│                      ┆                 ┆        ┆                      ┆                      ┆                      ┆ e1i68x64             │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ap ┆ WIN-KPO4DDU11AB ┆ 98     ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-AppxPackaging ┆ MS-Win-AppxPackaging │
│ pxPackaging%4Operati ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ /Op                  ┆ OM                   │
│ onal.evtx            ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Client-Lic ┆ WIN-KPO4DDU11AB ┆ 84     ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Cli-Licensing-Pla ┆ MS-Cli-Licensing-Pla │
│ ensing-Platform%4Adm ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ tform/Admin          ┆ tform                │
│ in.evtx              ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-St ┆ WIN-KPO4DDU11AB ┆ 30     ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:43:02. ┆ MS-Win-Storage-Storp ┆ MS-Win-StorPort      │
│ orage-Storport%4Heal ┆                 ┆        ┆ 260472Z              ┆ 939883Z              ┆ ort/Health           ┆                      │
│ th.evtx              ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Pr ┆ WIN-KPO4DDU11AB ┆ 29     ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-Privacy-Audit ┆ MS-Win-Privacy-Audit │
│ ivacy-Auditing%4Oper ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ ing/Op               ┆ ing                  │
│ ational.evtx         ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ke ┆ WIN-KPO4DDU11AB ┆ 16     ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:43:09. ┆ MS-Win-Kern-Cache/Op ┆ MS-Win-Kern-Cache    │
│ rnel-Cache%4Operatio ┆                 ┆        ┆ 260472Z              ┆ 408812Z              ┆                      ┆                      │
│ nal.evtx             ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Wi ┆ WIN-KPO4DDU11AB ┆ 15     ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:42:32. ┆ MS-Win-Win           ┆ Defender             │
│ ndows Defender%4Oper ┆                 ┆        ┆ 260472Z              ┆ 976932Z              ┆ Defender/Op          ┆                      │
│ ational.evtx         ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Te ┆ WIN-KPO4DDU11AB ┆ 12     ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:43:02. ┆ MS-Win-TermSvcs-Loca ┆ RDS-LSM              │
│ rminalServices-Local ┆                 ┆        ┆ 260472Z              ┆ 939883Z              ┆ lSessionMgr/Op       ┆                      │
│ SessionManager%4Oper ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
│ ational.evtx         ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-St ┆ WIN-KPO4DDU11AB ┆ 12     ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:21. ┆ MS-Win-Storage-Storp ┆ MS-Win-StorPort      │
│ orage-Storport%4Oper ┆                 ┆        ┆ 620626Z              ┆ 469910Z              ┆ ort/Op               ┆                      │
│ ational.evtx         ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-WM ┆ WIN-KPO4DDU11AB ┆ 11     ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-WMI-Activity/ ┆ WMI-Activity         │
│ I-Activity%4Operatio ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ Op                   ┆                      │
│ nal.evtx             ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Wi ┆ WIN-KPO4DDU11AB ┆ 10     ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:43:09. ┆ MS-Win-Win Firewall  ┆ Firewall             │
│ ndows Firewall With  ┆                 ┆        ┆ 260472Z              ┆ 408812Z              ┆ With Advanced        ┆                      │
│ Advanced Security%4F ┆                 ┆        ┆                      ┆                      ┆ Sec/Firewall         ┆                      │
│ irewall.evtx         ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Au ┆ WIN-KPO4DDU11AB ┆ 8      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-Audio/Op      ┆ MS-Win-Audio         │
│ dio%4Operational.evt ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆                      ┆                      │
│ x                    ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Gr ┆ WIN-KPO4DDU11AB ┆ 7      ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:43:02. ┆ MS-Win-GrpPolicy/Op  ┆ MS-Win-GrpPolicy     │
│ oupPolicy%4Operation ┆                 ┆        ┆ 260472Z              ┆ 939883Z              ┆                      ┆                      │
│ al.evtx              ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ap ┆ WIN-KPO4DDU11AB ┆ 7      ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:43:02. ┆ MS-Win-AppReadiness/ ┆ MS-Win-AppReadiness  │
│ pReadiness%4Admin.ev ┆                 ┆        ┆ 260472Z              ┆ 939883Z              ┆ Admin                ┆                      │
│ tx                   ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Wc ┆ WIN-KPO4DDU11AB ┆ 7      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:09. ┆ MS-Win-Wcmsvc/Op     ┆ MS-Win-Wcmsvc        │
│ msvc%4Operational.ev ┆                 ┆        ┆ 622783Z              ┆ 408812Z              ┆                      ┆                      │
│ tx                   ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Vo ┆ WIN-KPO4DDU11AB ┆ 6      ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:42:32. ┆ MS-Win-VolumeSnapsho ┆ MS-Win-VolumeSnapsho │
│ lumeSnapshot-Driver% ┆                 ┆        ┆ 260472Z              ┆ 976932Z              ┆ t-Driver/Op          ┆ t-Driver             │
│ 4Operational.evtx    ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ke ┆ WIN-KPO4DDU11AB ┆ 5      ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:42:17. ┆ MS-Win-Kern-WHEA/Op  ┆ MS-Win-Kern-WHEA     │
│ rnel-WHEA%4Operation ┆                 ┆        ┆ 260472Z              ┆ 080899Z              ┆                      ┆                      │
│ al.evtx              ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Nt ┆ WIN-KPO4DDU11AB ┆ 5      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-Ntfs/Op       ┆ MS-Win-Ntfs          │
│ fs%4Operational.evtx ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ap ┆ WIN-KPO4DDU11AB ┆ 4      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:09. ┆ MS-Win-AppModel-Runt ┆ MS-Win-AppModel-Runt │
│ pModel-Runtime%4Admi ┆                 ┆        ┆ 620810Z              ┆ 408812Z              ┆ ime/Admin            ┆ ime                  │
│ n.evtx               ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Sm ┆ WIN-KPO4DDU11AB ┆ 4      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-SmbCli/Conn   ┆ MS-Win-SMBCli        │
│ bClient%4Connectivit ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆                      ┆                      │
│ y.evtx               ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Cr ┆ WIN-KPO4DDU11AB ┆ 4      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-Crypto-DPAPI/ ┆ MS-Win-Crypto-DPAPI  │
│ ypto-DPAPI%4Operatio ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ Op                   ┆                      │
│ nal.evtx             ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Un ┆ WIN-KPO4DDU11AB ┆ 4      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-UnivTelemetry ┆ MS-Win-UnivTelemetry │
│ iversalTelemetryClie ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ Cli/Op               ┆ Cli                  │
│ nt%4Operational.evtx ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Bi ┆ WIN-KPO4DDU11AB ┆ 3      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-Biometrics/Op ┆ MS-Win-Biometrics    │
│ ometrics%4Operationa ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆                      ┆                      │
│ l.evtx               ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ke ┆ WIN-KPO4DDU11AB ┆ 3      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:21. ┆ MS-Win-Kern-ShimEngi ┆ MS-Win-Kern-ShimEngi │
│ rnel-ShimEngine%4Ope ┆                 ┆        ┆ 620626Z              ┆ 469910Z              ┆ ne/Op                ┆ ne                   │
│ rational.evtx        ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Us ┆ WIN-KPO4DDU11AB ┆ 3      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:21. ┆ MS-Win-UserPnp/Devic ┆ MS-Win-UserPnp       │
│ erPnp%4DeviceInstall ┆                 ┆        ┆ 620626Z              ┆ 469910Z              ┆ eInstall             ┆                      │
│ .evtx                ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-SM ┆ WIN-KPO4DDU11AB ┆ 2      ┆ 2022-03-01T16:42:16. ┆ 2022-03-01T16:42:17. ┆ MS-Win-SMBSvr/Op     ┆ MS-Win-SMBSvr        │
│ BServer%4Operational ┆                 ┆        ┆ 237770Z              ┆ 080899Z              ┆                      ┆                      │
│ .evtx                ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ta ┆ WIN-KPO4DDU11AB ┆ 2      ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:43:02. ┆ MS-Win-TaskScheduler ┆ TaskScheduler        │
│ skScheduler%4Mainten ┆                 ┆        ┆ 260472Z              ┆ 939883Z              ┆ /Maintenance         ┆                      │
│ ance.evtx            ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ke ┆ WIN-KPO4DDU11AB ┆ 2      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:09. ┆ MS-Win-Kern-Boot/Op  ┆ MS-Win-Kern-Boot     │
│ rnel-Boot%4Operation ┆                 ┆        ┆ 620810Z              ┆ 408812Z              ┆                      ┆                      │
│ al.evtx              ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Pa ┆ WIN-KPO4DDU11AB ┆ 2      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-Partition/Dia ┆ MS-Win-Partition     │
│ rtition%4Diagnostic. ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ gnostic              ┆                      │
│ evtx                 ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-AA ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:42:32. ┆ MS-Win-AAD/Op        ┆ MS-Win-AAD           │
│ D%4Operational.evtx  ┆                 ┆        ┆ 260472Z              ┆ 976932Z              ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Bi ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:42:32. ┆ MS-Win-BitLocker/Bit ┆ MS-Win-BitLocker-API │
│ tLocker%4BitLocker   ┆                 ┆        ┆ 260472Z              ┆ 976932Z              ┆ Locker Mgmt          ┆                      │
│ Management.evtx      ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Co ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:01. ┆ 2022-03-01T16:43:02. ┆ MS-Win-Containers-Bi ┆ MS-Win-Containers-Bi │
│ ntainers-BindFlt%4Op ┆                 ┆        ┆ 260472Z              ┆ 939883Z              ┆ ndFlt/Op             ┆ ndFlt                │
│ erational.evtx       ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-De ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:09. ┆ MS-Win-DeviceSetupMg ┆ MS-Win-DeviceSetupMg │
│ viceSetupManager%4Ad ┆                 ┆        ┆ 622783Z              ┆ 408812Z              ┆ r/Admin              ┆ r                    │
│ min.evtx             ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Co ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:09. ┆ MS-Win-Containers-Wc ┆ MS-Win-Containers-Wc │
│ ntainers-Wcifs%4Oper ┆                 ┆        ┆ 622783Z              ┆ 408812Z              ┆ ifs/Op               ┆ ifs                  │
│ ational.evtx         ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Dh ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:09. ┆ MS-Win-Dhcp-Cli/Admi ┆ MS-Win-Dhcp-Cli      │
│ cp-Client%4Admin.evt ┆                 ┆        ┆ 620810Z              ┆ 408812Z              ┆ n                    ┆                      │
│ x                    ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Re ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-ReadyBoost/Op ┆ MS-Win-ReadyBoost    │
│ adyBoost%4Operationa ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆                      ┆                      │
│ l.evtx               ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Ap ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-AppReadiness/ ┆ MS-Win-AppReadiness  │
│ pReadiness%4Operatio ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ Op                   ┆                      │
│ nal.evtx             ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Li ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-LiveId/Op     ┆ MS-Win-LiveId        │
│ veId%4Operational.ev ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆                      ┆                      │
│ tx                   ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Cr ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-Crypto-NCrypt ┆ MS-Win-Crypto-NCrypt │
│ ypto-NCrypt%4Operati ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ /Op                  ┆                      │
│ onal.evtx            ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-We ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-WebAuthN/Op   ┆ MS-Win-WebAuthN      │
│ bAuthN%4Operational. ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆                      ┆                      │
│ evtx                 ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-St ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-StorageSpaces ┆ MS-Win-StorageSpaces │
│ orageSpaces-Driver%4 ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ -Driver/Op           ┆ -Driver              │
│ Operational.evtx     ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Co ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:11. ┆ MS-Win-CodeIntegrity ┆ MS-Win-CodeIntegrity │
│ deIntegrity%4Operati ┆                 ┆        ┆ 620810Z              ┆ 893320Z              ┆ /Op                  ┆                      │
│ onal.evtx            ┆                 ┆        ┆                      ┆                      ┆                      ┆                      │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ Microsoft-Windows-Nt ┆ WIN-KPO4DDU11AB ┆ 1      ┆ 2022-03-01T16:42:00. ┆ 2022-03-01T16:43:21. ┆ MS-Win-Ntfs/WHC      ┆ MS-Win-Ntfs          │
Elapsed time: 00:00:08.990
```

fukusuket commented:
I created https://github.com/Yamato-Security/hayabusa/pull/1484#issuecomment-2469739234 issue as separate one.
- https://github.com/Yamato-Security/hayabusa/issues/1487

YamatoSecurity commented:
@fukusuket Thanks! It seems that `--timeline-offset` is not working. Can you check this?

YamatoSecurity commented:
Sorry, just a few small things.
1. Can you change the `¦` separator to `  ¦ ` with spaces around it?
2. Does the abbreviations first check the `channel_abbreviations.txt` file before `generic_abbreviations.txt`? I see `MS-Win-PwrShell/Op` but it should be `PwSh/Op`. Also, provider abbreviations should first check `provider_abbreviations.txt` and then after that use `generic_abbreviations.txt`.
3. Could you add a `-M, --multiline                    Output event field information in multiple rows` under `Output` that uses newlines instead of `  ¦ ` like we do for `csv-timeline`?

fukusuket commented:
@YamatoSecurity 
Thank you for checking! I fixed following points! Could you check it?🙏

>It seems that --timeline-offset is not working. Can you check this?
>1. Can you change the ¦ separator to  ¦ with spaces around it?
>2. Does the abbreviations first check the channel_abbreviations.txt file before generic_abbreviations.txt? I see MS-Win-PwrShell/Op but it should be PwSh/Op. Also, provider abbreviations should first check provider_abbreviations.txt and then after that use generic_abbreviations.txt.
>3. Could you add a -M, --multiline                    Output event field information in multiple rows under Output that uses newlines instead of  ¦ like we do for csv-timeline?

YamatoSecurity reviewed (approved):
@fukusuket LGTM! Thanks so much!
